java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -3356,6 +3357,18 @@ public CompletableFuture<String> publish(@NonNull String channel, @NonNull Strin
                 });
     }
 
+    @Override
+    public CompletableFuture<GlideString> publish(@NonNull GlideString message, @NonNull GlideString channel) {
```

Comment:
I think you can keep it to return `String` (same `set` command does)
```suggestion
    public CompletableFuture<String> publish(@NonNull GlideString message, @NonNull GlideString channel) {
```